File: src/main/java/com/zerozero/auth/application/OAuthRestClient.java
package com.zerozero.auth.application;

import com.zerozero.auth.infrastructure.oauth.core.OAuthAccessTokenResponse;
import com.zerozero.auth.infrastructure.oauth.core.OAuthResourceResponse;

import java.net.URI;

public interface OAuthRestClient {

    URI getAuthUrl();

    OAuthAccessTokenResponse getAccessToken(String authCode);

    OAuthResourceResponse getResource(String accessToken);
}

File: src/main/java/com/zerozero/auth/application/OAuthRestClient.kt
package com.zerozero.auth.application

import com.zerozero.auth.infrastructure.oauth.core.OAuthAccessTokenResponse
import com.zerozero.auth.infrastructure.oauth.core.OAuthResourceResponse
import java.net.URI

interface OAuthRestClient {
    val authUrl: URI

    fun getAccessToken(authCode: String): OAuthAccessTokenResponse?

    fun getResource(accessToken: String): OAuthResourceResponse?
}
